main.go

```diff
@@ -0,0 +1,258 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/blang/semver/v4"
+	_ "github.com/mattn/go-sqlite3"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type channelEntry struct {
+	packageName            string
+	channelName            string
+	bundleName             string
+	depth                  int
+	bundleVersion          string
+	bundleSkipRange        string
+	bundleSkip             string
+	replacesBundleName     string
+	headOperatorBundleName string
+	defaultChannel         string
+}
+
+type pkg struct {
+	name           string
+	bundles        map[string]*bundle
+	defaultChannel string
+}
+
+type bundle struct {
+	name            string
+	version         string
+	packageName     string
+	skips           string
+	skipRange       string
+	minDepth        int
+	channels        sets.String
+	replaces        sets.String
+	skipRanges      sets.String
+	isBundlePresent bool
+	channelHeads    sets.String
+}
+
+var indent1 = "  "
+var indent2 = "    "
+var indent3 = "      "
```

Comment:
why this intent is required? Could we use a better name on that?

Reply:
Done in 808fda7c